tools/windows/install-help/cal/customactiondata.cpp

```diff
@@ -302,73 +278,53 @@ void CustomActionData::findSuppliedUserInfo(std::wstring &input, std::wstring &c
 
 bool CustomActionData::parseUsernameData()
 {
-    std::wstring tmpName = ddAgentUserName;
-    bool previousInstall = findPreviousUserInfo();
-    bool userSupplied = false;
-
-    if (this->value(propertyDDAgentUserName, tmpName))
-    {
-        if (tmpName.length() == 0)
-        {
-            tmpName = ddAgentUserName;
-        }
-        else
-        {
-            userSupplied = true;
-        }
-    }
-    if (std::wstring::npos == tmpName.find(L'\\'))
-    {
-        WcaLog(LOGMSG_STANDARD, "loaded username doesn't have domain specifier, assuming local");
-        tmpName = L".\\" + tmpName;
-    }
-    // now create the splits between the domain and user for all to use, too
-    std::wstring computed_domain, computed_user;
+    auto userFromPreviousInstall= findPreviousUserInfo();
+    auto userFromCommandLine = findSuppliedUserInfo();
 
     // if this is an upgrade (we found a previously recorded username in the registry)
     // and nothing was supplied on the command line, don't bother computing that.  Just use
     // the existing
-    if (previousInstall && !userSupplied)
+    if (userFromPreviousInstall && !userFromCommandLine)
     {
-        computed_domain = pvsDomain;
-        computed_user = pvsUser;
         WcaLog(LOGMSG_STANDARD, "Using username from previous install");
+        _user = userFromPreviousInstall.value();
+    }
+    else if (userFromCommandLine)
+    {
```

Comment:
```suggestion
    {
        if (userFromPreviousInstall && userFromPreviousInstall.value() != userFromCommandLine.value())
        {
            WcaLog(LOGMSG_STANDARD, "Provided username doesn't match user from previous install, using provided one.");
        }
```
Let's log everything.